options/options.go

```diff
@@ -452,6 +461,60 @@ func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
 	return args, err
 }
 
+// ParseConfigFile iterates over args to find a --config option. If not found, we return.
+// If found, we read the contents of the specified config file in YAML format. We parse
+// any values corresponding to --password, --uri and --sslPEMKeyPassword, and store them
+// in the opts.
+func (opts *ToolOptions) ParseConfigFile(args []string) error {
+	// Get config file path from the arguments, if specified.
+	configPath := ""
+	for i, arg := range args {
+		if strings.HasPrefix(arg, "--config=") {
```

Comment:
Is there going to be a difference between `--config=foo` and `--config="foo"`? Will the latter make the config path be `\"foo\"`?

Reply:
Hmm, interesting point. It seems the underlying call to `os.Open()` is smart enough to recognize that the surrounding quotation marks aren't intended unless they're explicitly escaped. So there should be no difference.

From trying it on the command line as-is:
```
$ echo "uri: mongodb://abc:@localhost:33333" > config.yaml

$ mongodump --config=config.yaml
Enter password:

$ mongodump --config="config.yaml"
Enter password:

$ mongodump --config=\"config.yaml\"
2020-11-25T14:19:02.953-0500	error parsing command line options: open "config.yaml": no such file or directory

$ mongodump --config="\"config.yaml\""
2020-11-25T14:19:12.025-0500	error parsing command line options: open "config.yaml": no such file or directory
```

Reply:
I don't think it's the call to `os.Open` -- my understanding is that the shell consumes the quotes, so they are never passed as args in this specific example.

I think there are other issues here though. For example, does go-flags allow `/`-style flags on windows? If so, that's a whole separate class of thing that we have to manually check for here.

As I have noted in another PR, I'm generally skeptical of trying to match the library's flag-parsing behavior by manually scanning arguments